consensus/core/src/block_manager.rs

```diff
@@ -544,12 +532,92 @@ impl BlockManager {
         );
     }
 
+    // Tries to accept blocks that have been committed. Returns all the blocks that have been accepted, both from the ones
+    // provided and any children blocks.
+    #[tracing::instrument(skip_all)]
+    pub(crate) fn try_accept_committed_blocks(
```

Comment:
Is it simpler to write this new function, compared to adding branches to  `try_accept_blocks()`? There will be maintenance overhead for having two functions, but this approach may be cleaner.